src/DIRAC/StorageManagementSystem/Agent/StageRequestAgent.py

```diff
@@ -410,7 +410,7 @@ def __addAssociatedReplicas(self, replicasToStage, seReplicas, allReplicaInfo):
       seReplicas.setdefault(storageElement, []).extend(seReplicaIDs)
       replicasToStage.extend(seReplicaIDs)
 
-    for replicaID in allReplicaInfo.keys():
+    for replicaID in allReplicaInfo:
```

Comment:
As elements are deleted during iteration:
```suggestion
    for replicaID in list(allReplicaInfo):
```

Reply:
Done.